tictactoe/models/Board.java

```diff
@@ -0,0 +1,97 @@
+package tictactoe.tictactoePackage.models;
+
+public class Board {
+
+    private int sizeOfBoard;
+    private Character  board[][];
+
+    public Board(int sizeOfBoard){
+        this.sizeOfBoard = sizeOfBoard;
+        this.board = new Character [sizeOfBoard][sizeOfBoard];
+    }
+
+    public void setPieceAtBoard(int currentRow, int currentColumn, Character  currentPiece) {
+        if(isPieceAtValidCell(currentRow, currentColumn)){
+            board[currentRow][currentColumn] = currentPiece;
+        }
+        else{
+            System.out.println("Invalid Move");
+        }
+    }
+
+    public void initializeTheBoard(){
+        for (int i=0;i<sizeOfBoard;i++){
+            for (int j=0;j<sizeOfBoard;j++){
+                board[i][j] = '-';
+            }
+        }
+    }
+
+    public void printBoard(){
+        for (int i=0;i<sizeOfBoard;i++){
+            for (int j=0;j<sizeOfBoard;j++){
+                System.out.print(board[i][j]);
+            }
+            System.out.println();
+        }
+    }
+
+    public boolean isPieceAtValidCell(int currentRow, int currentColumn){
+        if(currentRow<0 || currentRow>= sizeOfBoard) {
+            return false;
+        }
+        if(currentColumn<0 || currentColumn>= sizeOfBoard) {
+            return false;
+        }
+        if(board[currentRow][currentColumn] != '-') return false;
+        return true;
+    }
+
+    public boolean HasPlayerWonTheGame(int currentRow, int currentColumn){
+
```

Comment:
Can be broken down to smaller functions

Reply:
Because I cannot understand what each line means here. Can be abstracted to make it easily understandable.